src/main/resources/public/ts/directives/calendar-item/calendar-item.ts

```diff
@@ -59,25 +59,42 @@ class Controller implements ng.IController, IViewModel {
         if($event) $event.stopPropagation();
         this.loading = true;
         safeApply(this.$scope);
-        this.calendarService.updateExternalCalendar(this.$scope.vm.calendar);
-        this.$timeout(() : IPromise<void> => {
-            this.calendarService.checkExternalCalendarSync(this.$scope.vm.calendar)
-                .then((r:AxiosResponse) => {
-                    if(r.data['isUpdating'] == false) {
+        try {
+            await this.calendarService.updateExternalCalendar(this.$scope.vm.calendar);
+            this.$timeout(() : IPromise<void> => {
+                this.calendarService.checkExternalCalendarSync(this.$scope.vm.calendar)
+                    .then((r:AxiosResponse) => {
+                        if(r.data['isUpdating'] == false) {
```

Comment:
Obligé de faire ['isUpdating']? ton data est généralement en mode "any" donc tu peux faire directement r.data.isUpdating

Une autre alternative serait de typer ton data :  
```suggestion
                        if((<ton_type>res.data).isUpdating == false) {
```

Reply:
le truc c'est que data a le type any

Reply:
Rien ne t'empêche de le typer / caster !